File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/ledger/RevocGetRegRequestTest.java
package org.hyperledger.indy.sdk.ledger;

import org.hyperledger.indy.sdk.utils.PoolUtils;
import org.json.JSONObject;
import org.junit.Test;

import java.util.Date;

public class RevocGetRegRequestTest extends LedgerIntegrationTest {

	@Test
	public void testBuildGetRevocRegRequestWorks() throws Exception {
		JSONObject expectedResult = new JSONObject()
				.put("operation", new JSONObject()
						.put("type", "116")
						.put("revocRegDefId", revRegDefId)
						.put("timestamp", 100)
				);

		String request = Ledger.buildGetRevocRegRequest(DID, revRegDefId, 100).get();

		assert (new JSONObject(request).toMap().entrySet()
				.containsAll(
						expectedResult.toMap().entrySet()));
	}

	@Test
	public void testRevocRegRequestsWorks() throws Exception {
		postEntities();

		int timestamp = (int) (new Date().getTime()/1000) + 100;

		String getRevRegRequest = Ledger.buildGetRevocRegRequest(DID, revRegDefId, timestamp).get();
		String getRevReResponse = PoolUtils.ensurePreviousRequestApplied(pool, getRevRegRequest, response -> {
			JSONObject responseObject = new JSONObject(response);
			return !responseObject.getJSONObject("result").isNull("seqNo");
		});

		Ledger.parseGetRevocRegResponse(getRevReResponse).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/LedgerTests/RevocGetRegRequestTest.cs
using Hyperledger.Indy.LedgerApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.LedgerTests
{
    [TestClass]
    public class RevocGetRegRequestTest : LedgerIntegrationTestBase
    {
        [TestMethod]
        public async Task TestBuildGetRevocRegRequestWorks()
        {
            var expectedResult =
                "\"operation\":{" +
                "\"type\":\"116\"," +
                "\"revocRegDefId\":\"RevocRegID\"," +
                "\"timestamp\":100" +
                "}";

            var request = await Ledger.BuildGetRevocRegRequestAsync(DID, "RevocRegID", 100);

            Assert.IsTrue(request.Replace("\\s+", "").Contains(expectedResult.Replace("\\s+", "")));
        }

        [TestMethod]
        public async Task TestRevocRegRequestsWorks()
        {
            await PostEntitiesAsync();

            var timespan = (DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc));
            var timestamp = (int)timespan.TotalSeconds + 100;

            var getRevRegRequest = await Ledger.BuildGetRevocRegRequestAsync(DID, revRegDefId, timestamp);
            var getRevRegResponse = await PoolUtils.EnsurePreviousRequestAppliedAsync(pool, getRevRegRequest, response => {
                var responseObject = JObject.Parse(response);
                return responseObject["result"]["seqNo"] != null;
            });

            await Ledger.ParseGetRevocRegResponseAsync(getRevRegResponse);
        }
    }
}
